### PR TITLE
New nav design with support for groups and configurable subcomponents

### PR DESCRIPTION
**Pull request by mojavelinux**
This PR implements the new design for the site navigation based on the design comp in Figma. Aside from the design refresh, the main feature is the reorganization of the components (aka products) into groups. For the UI preview, these groups are defined under the `nav` key in the ui-model.yml file. For the real site, this configuration will be read from a file named antora-nav.yml (or similar). The same configuration is also used to define subcomponents, such as the components grouped under the connectors component.

Here's how that configuration looks:

```yml
nav:
  subcomponents:
  - parent: connectors
    components: '*-connector'
  groups:
  - root: true
    components: [home, release-notes]
  - title: Getting Started
    components: general
  - title: Docs by Product
    components: ['*', tcat-server]
```

This configuration replaces the `site-component-order` attribute in the Antora playbook.

Rather than try to patch support for groups and configurable subcomponents into the existing nav script (01-nav.js), we made the decision to rewrite the script from scratch. The rewritten code has been designed to be functional so that it's easier to follow and maintain. It also addresses numerous outstanding bugs, such as the height of the nav panel exceeding the visible area.

Here are the notable changes this PR introduces:

* apply new nav design from design comp in figma
* stylize scrollbars on nav, removing track, making thumb width thin, and only showing on hover
* add support for navigation groups
* drop the group title for the root-level group
* drop the component title when a group only contains a single, unversioned component
* use configuration to control subcomponent grouping (such as connectors and rtfcloud)
* indent subcomponents in nav and use smaller icon to indicate their secondary position
* adjust height of nav so it always fits in the available space on the screen (desktop and mobile); don't allow nav to extend off screen
* add animation when sliding nav in and out on mobile
* fix position of nav and toc relative to banner when banner is present
* fix z-index of banner so it appears below the box shadow of the header
* only construct the nav list for each component version on demand
* activate nav version menu panel without using the tippy library
* remove touchend event listeners on nav toggle so nav is more easily activated on a touch device
* remove all touchend event listeners as they are unnecessary
* remove unused CSS vars (remove base.css)
* fix mapping for bold font weight (should be 700 instead of 600; adjust font-face to match)
* set page-version meta to real version instead of display version
* add missing DINPro bold font (same change that was made in PR for landing page)
* bind preview server to local IP address for testing from another device (such as a phone)
* don't load Google Analytics in UI preview if GOOGLE_ANALYTICS_KEY is not set

### CODE REVIEW

**Comment by mojavelinux** [comment]
If you prefer, I can squash these commits before merge.

File: NOTICE.adoc [hunk 1]
```diff
@@ -0,0 +1,21 @@
+= Notices for the MuleSoft Documentation UI Project
+
+This UI for Antora is produced and maintained by the MuleSoft Documentation team.
+
+== Copyright
+
+This UI is derived from the https://gitlab.com/antora/antora-ui-default/[Antora Default UI].
+Modifications have been applied to align the theme and behavior of the UI with the MuleSoft brand and documentation requirements.
+All branding elements in this UI, including icons, images, and fonts, are copyright Salesforce.
```
Comment (graphitefriction) [inline NOTICE.adoc:9]: I'd change this to:

MuleSoft brand elements, including icons, images, and fonts, and the visual design of this UI are copyright MuleSoft LLC, a Salesforce company. All rights to the brand elements and visual design are reserved and not licensed under the terms mentioned below.

File: NOTICE.adoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+== Declared Project Licenses
```
Comment (graphitefriction) [inline NOTICE.adoc:11]: I'd move this section below the section Source Code, nest it (`===`), and change the title to "Source Code Licenses"

File: NOTICE.adoc [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+As a derivative work of the Antora Default UI project, the source code in this repository is licensed under the terms of the Mozilla Public License 2.0 (MPL-2.0).
```
Comment (graphitefriction) [inline NOTICE.adoc:13]: Add a link to the text of the license

**Comment by cheungaryk** [comment]
@mojavelinux do you mind resolving the conflicts?

**Comment by cheungaryk** [comment]
Thanks for resolving the merge conflicts. I have manually created a release https://github.com/mulesoft/docs-site-ui/releases/tag/test-1 for @Cristian-Venticinque to test on an internal beta site

**Comment by hannanelson** [comment]
> Thanks for resolving the merge conflicts. I have manually created a release https://github.com/mulesoft/docs-site-ui/releases/tag/test-1 for @Cristian-Venticinque to test on an internal beta site

@cheungaryk - I'm interested in how to use this for a POC I'm working on.

**Comment by mojavelinux** [comment]
Remember that you can also override the UI bundle location using a CLI option when calling Antora:

```
--ui-bundle-url=https://github.com/mulesoft/docs-site-ui/releases/download/test-1/ui-bundle.zip
```

**Comment by cheungaryk** [comment]
this was merged into another branch, and with some cleaning up, merged into master and deployed to the actual site (https://docs.mulesoft.com/). So we can close it - thank you!

**Comment by mojavelinux** [comment]
I'm about a month late, but I still wanted to say congratulations on rolling out this awesome new feature! It looks great!

**Pull request closed without merging**